src/libraries/System.Private.CoreLib/src/System/Environment.Windows.cs

```diff
@@ -122,6 +122,28 @@ public static string SystemDirectory
             }
         }
 
+        public static bool UserInteractive
+        {
+            get
+            {
+                // Per documentation of GetProcessWindowStation, this handle should not be closed
+                IntPtr handle = Interop.User32.GetProcessWindowStation();
+                if (handle == IntPtr.Zero)
```

Comment:
I think that Nano should be rather treated as Unix, and always return true.

I would delete this special early out.

Reply:
Note that the improper use of this API for detecting whether the process is running as Windows Service is not going to work on Nano in any case.